module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -463,10 +463,24 @@ public function getConfig($function, $params = [])
      */
     protected function isHoldable($locationName)
     {
-        return !in_array(
-            $locationName,
-            (array)($this->config['Holds']['excludeHoldLocations'] ?? [])
-        );
+        $mode = $this->config['Holds']['excludeHoldLocationsCompareMode'] ?? 'exact';
+        $excludeLocs = (array)$this->config['Holds']['excludeHoldLocations'] ?? [];
+
+        // Exclude checking by regex match
+        if ($mode == "regex") {
+            foreach ($excludeLocs as $pattern) {
+                if (preg_match($pattern, $locationName) === 1) {
+                    return false;
+                }
+            }
+            return true;
+        } else {
+            // Otherwise exclude checking by exact match
+            return !in_array(
```

Comment:
You can probably collapse this function call into a single line -- I think it was spread out before because $excludeLocs used to be a long expression, but it should be readable in a more compact presentation now.